### PR TITLE
fix github links

### PR DESCRIPTION
**Pull request by ardrabczyk**
As discussed at https://lists.slackbuilds.org/pipermail/slackbuilds-users/2024-July/029559.html

### CODE REVIEW

**Comment by aclemons** [comment]
@sbo-bot: rebuild network/ngrep

**Comment by aclemons** [comment]
@sbo-bot: rebuild system/dstat

**Comment by aclemons** [comment]
@sbo-bot: rebuild network/radicale

**Comment by ardrabczyk** [comment]
I built radicale using sbopkg on x86-64 here and it worked, I'll check what's going on.

File: network/radicale/radicale.SlackBuild [hunk 1]
```diff
@@ -90,7 +90,7 @@ rm -rf $PKG
 mkdir -p $TMP $PKG $OUTPUT
 cd $TMP
 rm -rf $PRGNAM-$VERSION
-tar xvf $CWD/$PRGNAM-$VERSION.tar.gz || tar xvf $CWD/Radicale-$VERSION.tar.gz || tar xvf v$VERSION.tar.gz
+tar xvf $CWD/$PRGNAM-$VERSION.tar.gz
```
Comment (aclemons) [inline network/radicale/radicale.SlackBuild:93]: The name of the project is actually `Kozea/Radicale` - the current link needs to change:

from https://github.com/SlackBuildsOrg/slackbuilds/blob/cdef13221b8fac6b940471ce1841accae367eb2d/network/radicale/radicale.info#L4C11-L4C91

It should be:

https://github.com/Kozea/Radicale/archive/refs/tags/v3.1.8/Radicale-3.1.8.tar.gz

Comment (ardrabczyk) [reply network/radicale/radicale.SlackBuild:93]: hmm not sure, lowercase version should work too

Comment (ardrabczyk) [reply network/radicale/radicale.SlackBuild:93]: also, I don't understand how it works, it says that radicale-3.1.8.tar.gz does not exist but then it builds the package?

Comment (willysr) [reply network/radicale/radicale.SlackBuild:93]: based on our SBo Github Guide, name of the tarball should match with the project name to give consistent output

Comment (ardrabczyk) [reply network/radicale/radicale.SlackBuild:93]: But is it related to the build failure? If I changed a link to https://github.com/Kozea/Radicale/archive/refs/tags/v3.1.8/Radicale-3.1.8.tar.gz I get:

    $ sudo env MAKEFLAGS='-j24 -l13' nice -n 5 bash ./radicale.SlackBuild
    tar: /home/ja/slackbuilds/network/radicale/radicale-3.1.8.tar.gz: Cannot open: No such file or directory
    tar: Error is not recoverable: exiting now

I want to know why did the build fail.

Comment (ardrabczyk) [reply network/radicale/radicale.SlackBuild:93]: so the build robot does not download files specified in .info files?

Comment (ardrabczyk) [reply network/radicale/radicale.SlackBuild:93]: I think that robot uses this for building https://github.com/aclemons/slackbuilds-ci/blob/master/.slackbuilds-library/resources/buildpkg#L1, I'll try to understand how it works.

Comment (aclemons) [reply network/radicale/radicale.SlackBuild:93]: > I want to know why did the build fail.

The build is run and downloads the links from the info file with content-disposition - just like it you navigate to the link on the SBo website or open the link in your browser.

So if you take the link currently in the info file:

https://github.com/Kozea/Radicale/archive/refs/tags/v3.1.8/radicale-3.1.8.tar.gz

open it firefox or chromium, you get Radicale-3.1.8.tar.gz.

But you removed handling for that filename from the SlackBuild. That is why the build currently fails.

> also, I don't understand how it works, it says that radicale-3.1.8.tar.gz does not exist but then it builds the package?

But it does fail:

```
Running radicale.SlackBuild ...                                     ETA 21:16:??
env MAKEFLAGS='-j24 -l13' nice -n 5 bash ./radicale.SlackBuild
network/radicale: radicale.SlackBuild failed (status 2)                 21:16:06
tar: /tmp/SBo/repo.7jb9St/slackbuild_radicale/radicale-3.1.8.tar.gz: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
See /var/log/repo/SBo/network/radicale/rebuild.log
Unmounting chroot ... done.
⛔️ network/radicale FAILED ⛔️
```

The github link guide is there to make it so you get the same filename whether you download it this way with a browser or anything that respects the content-disposition, or if you directly wget it and have wget us the last URI part as the filename.

Comment (ardrabczyk) [reply network/radicale/radicale.SlackBuild:93]: ok, thx for explanation. I believe that the robot goes on to using https://github.com/aclemons/slackrepo, I just wanted to understand what it does.

Comment (ardrabczyk) [reply network/radicale/radicale.SlackBuild:93]: oh no, I'll run linters again...

**Comment by willysr** [comment]
@ardrabczyk you don't need to rename the script, you just need to adjust the tarball link
renaming is normally done using new submission and remove the old, which is too much work
let's just keep the script name and use correct link in DOWNLOAD part
I will merge the first 2 commits and close this PR
please submit a new PR for radicale

**Comment by ardrabczyk** [comment]
> @ardrabczyk you don't need to rename the script, you just need to adjust the tarball link

but I also need to change `PRGNAM=radicale` to `PRGNAM=Radicale` in radicale.SlackBuild and then I get sbolint errors:

    [ja:~/slackbuilds/network/radicale] radicale-just-link+* ± sbolint .
    radicale: ERR: radicale.SlackBuild:27: PRGNAM doesn't match dir name (Radicale != radicale)
    radicale: NOTE: radicale.SlackBuild:66: SLKCFLAGS gets set, but never used.
    radicale: NOTE: radicale.SlackBuild:67: LIBDIRSUFFIX gets set, but never used.
    sbolint: radicale: errors 1, warnings 0

and this error when package is built:

    cat: /var/lib/sbopkg/local/network/radicale/Radicale.SlackBuild: No such file or directory

File: network/radicale/radicale.SlackBuild (hunk 1 above)
Comment (aclemons) [reply network/radicale/radicale.SlackBuild:93]: > ok, thx for explanation. I believe that the robot goes on to using https://github.com/aclemons/slackrepo, I just wanted to understand what it does.

Yes that's correct. The comments hit a github webhook which triggers a jenkins job which runs slackrepo to build the package. Jenkins then comments on this PR with the results.

**Pull request closed without merging**